[Tan Jia Jun] iP

tanjiajiajun opened the pull request:
# Dukepro - your local friendly chatbot
> "Out of clutter, find simplicity. From discord, find harmony. In the middle of difficulty lies opportunity." - Albert Einstein

NewDuke is a revolutionary task management chatbot that helps you track the progress of your work. It's
- Text-based
- Easy to learn 😸
- Fast


### All you need to do is:
1. Download it from [here](https://github.com/tanjiajiajun/ip)
2. Double click it
3. Add in your tasks
4. Let it manage your tasks for you!

Best of all, this all is **FREE!**

### Features
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Graphical User Interface (coming soon)


### Code - here's the `main` method 
```java
// Sample Java code with syntax highlighting
public class NewDuke {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

tanjiajiajun committed:
Level 0: Rename, Greet, Exit

tanjiajiajun committed:
Level 1. Echo

tanjiajiajun committed:
Level 2. Add, List

tanjiajiajun committed:
changed goodbye message

tanjiajiajun committed:
Level-3 Mark as Done

tanjiajiajun committed:
Added todos, events, deadlines

tanjiajiajun committed:
handled Unknown Imput ecceptions

tanjiajiajun committed:
handled empty todo, events and deadlines

tanjiajiajun committed:
allow delete

tanjiajiajun committed:
add automated text UI testing

tanjiajiajun committed:
Allow save and load of data

tanjiajiajun committed:
Merge branch 'branch-Level-7'

tanjiajiajun committed:
Convert dates from strings to Java DateTime Objects

tanjiajiajun committed:
Merge branch 'branch-Level-8'

tanjiajiajun committed:
Used more OOP

tanjiajiajun committed:
Add code to duke package

tanjiajiajun committed:
Merge remote-tracking branch 'origin/add-gradle-support'

tanjiajiajun committed:
Set up gradle so that you can build and run Duke using gradle

tanjiajiajun committed:
Add JUnit test for Storage and Todos class

tanjiajiajun committed:
Add ability for users a way to find a task by searching for a keyword.

tanjiajiajun committed:
Add JavaDoc header comments to non-private classes/methods.

tanjiajiajun committed:
Rewrite code to comply with coding standard

tanjiajiajun committed:
Resolve merge conflicts

tanjiajiajun committed:
Merge branch 'branch-A-CodingStandard'

tanjiajiajun committed:
Add JavaDoc for find method

tanjiajiajun committed:
Add checkstyle to standardise coding style

BryanL2303 commented on src/main/java/duke/Deadlines.java:
```diff
@@ -0,0 +1,20 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadlines extends Task {
```
I believe that majority of Java classes should not be plural? The coding standard does not explicitly state this for class names. But just like the parent class Task, I think it might make more sense for Deadline to be singular. Like wise for some of the other classes.

BryanL2303 commented on src/main/java/duke/DukeExceptions.java:
```diff
@@ -0,0 +1,10 @@
+package duke;
+
+public class DukeExceptions extends Exception{
```
I have done a little bit of reading, I think this would be one case where the plural name for the class is okay since you are inheriting from DukeExceptions. So there are a lot of DukeExceptions.

tanjiajiajun replied:
Thank you! Will amend in the next version.

BryanL2303 commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,95 @@
+/**
+ * The UI class handles user interface interactions.
+ */
+public class UI {
```
I believe the letter I should not be capitalized since UI is an abbreviation for User Interface.

tanjiajiajun replied:
Thank you! Will amend in the next version.

maiyasaliha replied:
I agree, Ui would follow the coding standard better!

BryanL2303 reviewed (commented):
Hi, I am assigned to do a PR review for your naming. There are only a few small potential issues that I have noted down.

tanjiajiajun committed:
Implement JavaFX GUI

maiyasaliha commented on src/main/java/commands/Commands.java:
```diff
@@ -0,0 +1,132 @@
+package commands;
+
+import duke.*;
+import exceptions.*;
```
I believe the coding standard advises against wildcard imports. Not that major maybe just take note of this!

maiyasaliha commented on src/main/java/commands/Commands.java:
```diff
@@ -0,0 +1,132 @@
+    public static ToDos todosCommand (String input, Storage storage) throws EmptyTaskException {
+        List<String> inputParts = Arrays.asList(input.split(" "));
+        String descriptor = "";
+        for (int i=1; i<inputParts.size(); i++) {
```
```suggestion
        for (int i = 1; i < inputParts.size(); i++) {
```

maiyasaliha replied:
Just take note of the spacing, I've noticed the same issue in several other places too.

maiyasaliha commented on src/main/java/commands/Commands.java:
```diff
@@ -0,0 +1,132 @@
+    public static String findCommand (String input, Storage storage) throws EmptyStringException {
+        List<String> inputParts = Arrays.asList(input.split(" "));
+        String identifier = "";
+        for (int j=1; j<inputParts.size(); j++) {
```
```suggestion
        for (int j = 1; j < inputParts.size(); j++) {
```

maiyasaliha commented on src/main/java/commands/Commands.java:
```diff
@@ -0,0 +1,132 @@
+        return e;
+    }
+
+
```
Very nicely done! The code is very easy to read!

maiyasaliha reviewed (commented):
Great Job on this! You have been on task and have completed everything successfully. Just a few minor coding standard violations. Do make sure to edit out the large chunk of commented code as well. But overall LGTM!

tanjiajiajun committed:
Add JavaDoc for Commands.java

tanjiajiajun committed:
Add assertion features

Introduce the use of assertions in the codebase to document critical assumptions that should hold at various execution points.

Used assertions to
* ensure that task's description cannot be empty.
* input cannot be empty.

tanjiajiajun committed:
Rename Todo, Deadline and Event java files

tanjiajiajun committed:
Add JavaDoc comments

tanjiajiajun committed:
Refactor Save.java to reduce nesting

tanjiajiajun committed:
Refactor: Simplify Nested Code Structures and Enhance JavaDoc Documentation

In this commit, we simplified the deeply nested code present in Save.java, as well as renamed classes such as Todo, Event and Deadline. We also added more JavaDoc Documentation to our code.

Merge branch 'branch-A-CodeQuality'

tanjiajiajun committed:
Fix Checkstyle

tanjiajiajun committed:
Add more checkstyle

tanjiajiajun committed:
Want to checkout

tanjiajiajun committed:
Ensure Code Quality

tanjiajiajun committed:
Merge branch 'master' into branch-A-CodeQuality

tanjiajiajun committed:
Merge pull request #2 from tanjiajiajun/branch-A-CodeQuality

Ensure Code Quality

tanjiajiajun committed:
Allow users to view their schedule at any date.

tanjiajiajun committed:
Finalise features for submission

tanjiajiajun committed:
Update JUnit Tests

tanjiajiajun committed:
update README

tanjiajiajun committed:
Update README.md

tanjiajiajun committed:
Update README.md

tanjiajiajun committed:
Add AI Assistance
